indico/modules/designer/util.py

```diff
@@ -38,3 +43,24 @@ def get_all_templates(obj):
 def get_inherited_templates(obj):
     """Get all templates inherited by a given event/category"""
     return get_all_templates(obj) - set(obj.designer_templates)
+
+
+def get_not_deletable_templates(obj):
+    """Get all non-deletable templates usable by an event/category"""
+
+    not_deletable_criteria = [
+        DesignerTemplate.is_system_template,
+        DesignerTemplate.backside_template_of != None,
+        DesignerTemplate.ticket_for_regforms.any(RegistrationForm.event_new.has(Event.ends_after(now_utc())))
+    ]
+    return set(DesignerTemplate.query.filter(DesignerTemplate.event_new == obj if isinstance(obj, Event) else
```

Comment:
Wrong indentation style.
```
return set(DesignerTemplate.query
           .filter(...)
           .whatever(...))
```

Also, I would really split this and move the statement with the isinstance check out of the query and assign it to a variable first, and then do e.g. `.filter(obj_criterion)`